### DISCUSSION TITLE
Fehler Kompiler nach Update Arduino Board Verwalter Version 3.0.0

### DISCUSSION THREAD

**Discussion by beabel** [beabel/radiowecker#50] [Fragen & Antworten / Q&A] [answered]
### Ab Version 3.0.0 kommt dieser Fehler:

`In file included from c:\Users\User\Documents\Arduino\libraries\ESP8266Audio\src\AudioFileSourceHTTPStream.cpp:23:
c:\Users\User\Documents\Arduino\libraries\ESP8266Audio\src\AudioFileSourceHTTPStream.h:56:5: error: 'WiFiClient' does not name a type
   56 |     WiFiClient client;`

~~Bisher habe ich noch keine Lösung außer auf Version 2.0.17 zurück gehen, dort funktioniert noch alles.~~

Dann muss die ESP8266Audio Bibliothek mind. 
### Version 1.9.9 
haben, danach sollte es wieder gehen.

> [!IMPORTANT]
> maximal die Version 3.0.0 sonst wird der sketch zu groß.

|Boardverwalter Version|Sketchgröße|
| ------------- | ------------- |
|2.0.17|:white_check_mark: 1113105|
|3.0.0|:white_check_mark: 1288101|
|3.0.5|:x: 1312869|

**Comment by beabel** [comment]
auf Version 2.0.17 zurück gehen, dort funktioniert noch alles.

**Reply by DarnokHillstar** [reply]
Hi,

ich habe jetzt fast 2 Tage versucht den Fehler zu finden!
Du bist spitze!
Danke

DarnokHillstar

**Comment by beabel** [comment]
# breaking changes in the API esp32 3.0.0

Dies scheinen die Problem Baustellen zu sein:

## I2S
> [I2S](https://docs.espressif.com/projects/arduino-esp32/en/latest/migration_guides/2.x_to_3.0.html#i2s)
The I2S driver has been completely redesigned and refactored to use the new ESP-IDF driver. For more information about the new API, check [I2S](https://docs.espressif.com/projects/arduino-esp32/en/latest/api/i2s.html).

## WIFI

> [WiFi](https://docs.espressif.com/projects/arduino-esp32/en/latest/migration_guides/2.x_to_3.0.html#wifi)
Functional changes
In Arduino (and other frameworks) the method named flush() is intended to send out the transmit buffer content. WiFiClient and WiFiUDP method flush() won’t clear the receive buffer anymore. A new method called clear() is now used for that. Currently flush() does nothing in WiFiClient, WiFiClientSecure and WiFiUDP.
WiFiServer has functions accept() and available() with the same functionality. In Arduino, available() should work differently so it is now deprecated.
WiFiServer had unimplemented write functions inherited from Print class. These are now removed. Also unimplemented method stopAll() is removed. The methods were unimplemented because WiFiServer doesn’t manage connected WiFiClient objects for print-to-all-clients functionality.

**Comment by beabel** [comment, marked as answer]
### Ab sofort kann man wieder updaten.

Wenn man die **ESP8266Audio** Bibliothek **Version 1.9.9** installiert, 
kann man danach auch den Boardverwalter esp32 > Version 3 installieren.
Es kommen keine compiler Fehler mehr.

### HINWEIS!!!


> [!IMPORTANT]
> maximal die Version 3.0.0 sonst wird der sketch zu groß.

|Boardverwalter Version|Sketchgröße|
| ------------- | ------------- |
|2.0.17|:white_check_mark: 1113105|
|3.0.0|:white_check_mark: 1288101|
|3.0.5|:x: 1312869|
